lib/matplotlib/path.py

```diff
@@ -341,6 +342,16 @@ def make_compound_path(cls, *args):
                 codes[i:i + len(path.codes)] = path.codes
             i += len(path.vertices)
 
+        # remove internal STOP's, replace kinal stop if present
+        last_vert = None
+        if codes.size > 0 and codes[-1] == cls.STOP:
+            last_vert = vertices[-1]
+        vertices = vertices[codes != cls.STOP, :]
+        codes = codes[codes != cls.STOP]
```

Comment:
For better readability and a minor performance gain, I'm slightly inclined to
```suggestion
        no_stop_mask = codes != cls.STOP
        vertices = vertices[no_stop_mask, :]
        codes = codes[no_stop_mask]
```

Reply:
incorporated manually